server/routes/charts.js

```diff
@@ -39,9 +40,45 @@ router.route('/charts/new')
 router.route('/charts/:chart_id')
   .get(ensureAuthenticated, async(req,res) => {
     try {
+      const data = []
+      const { dataDb, appDb } = req.app.locals
       const { chart_id } = req.params
+      const { access } =  await appDb
+        .collection(collections.users)
+        .findOne(
+          { uid: req.user },
+          { _id: 0, access: 1 }
+        )
+      const { assessment, variable, fieldLabelValueMap, title } = await dataDb
+        .collection(collections.charts)
+        .findOne({ _id: new ObjectID(chart_id) })
+
+      for await (const field of fieldLabelValueMap) {
+        const valueCountData = []
+
+        for await (const study of access ){
+          let count = 0;
+          const { label, value } = field;
+          const subjectList = await dataDb
+            .collection(collections.toc)
+            .find({ assessment, study }, { projection: { collection: 1, subject: 1, study: 1, _id: 0 }})
+            .map(doc => ({ ...doc, variable }))
+            .toArray()
+
+            for await (const doc of subjectList) {
+              const { collection, study, variable, subject } = doc
+              const counted = await dataDb
+                .collection(collection)
+                .findOne({ site: study, [variable]: value, subject_id: subject }, { projection: { [variable]: 1, _id: 0 }})
+              count = counted ? count+=1 : count
+            }
+            
+            valueCountData.push({ count, siteName: study, fieldLabel: label })
+          }
+          data.push(valueCountData)
+        }
```

Comment:
It's been a while since I've mongo'd but ☝️ this seems like it might be a good use case for [aggregation](https://mongodb.github.io/node-mongodb-native/3.6/api/Collection.html#aggregate)? Let me know if you wanna pair on that :)

Reply:
[3d5d8c](https://github.com/AMP-SCZ/dpdash/pull/91/commits/3d5d8c2367dffb4b84223b17754ce6f1a2ecb5c6)